clarifai/client/model.py

```diff
@@ -47,10 +57,243 @@ def __init__(self,
+      self.load_info()
+    if self.model_info.model_type_id not in TRAINABLE_MODEL_TYPES:
+      raise UserError(f"Model type {self.model_info.model_type_id} is not trainable")
+    request = service_pb2.ListModelTypesRequest(user_app_id=self.user_app_id,)
+    response = self._grpc_request(self.STUB.ListModelTypes, request)
+    if response.status.code != status_code_pb2.SUCCESS:
+      raise Exception(response.status)
+    templates = response_to_templates(
+        response=response, model_type_id=self.model_info.model_type_id)
+
+    return templates
+
+  def get_params(self, template: str = None, yaml_file: str = 'params.yaml') -> Dict[str, Any]:
+    """Returns the model params for the model type and yaml file.
+
+    Args:
+        template (str): The template to use for the model type.
+        yaml_file (str): The yaml file to save the model params.
+
+    Returns:
+        params (Dict): Dictionary of model params for the model type.
+
+    Example:
+        >>> from clarifai.client.model import Model
+        >>> model = Model(model_id='model_id', user_id='user_id', app_id='app_id')
+        >>> model_params = model.get_params(template='template', yaml_file='model_params.yaml')
+    """
+    if not self.model_info.model_type_id:
+      self.load_info()
+    if self.model_info.model_type_id not in TRAINABLE_MODEL_TYPES:
+      raise UserError(f"Model type {self.model_info.model_type_id} is not trainable")
+    if template is None and self.model_info.model_type_id not in [
+        "clusterer", "embedding-classifier"
+    ]:
+      raise UserError(
+          f"Template should be provided for {self.model_info.model_type_id} model type")
+    if template is not None and self.model_info.model_type_id in [
+        "clusterer", "embedding-classifier"
+    ]:
+      raise UserError(
+          f"Template should not be provided for {self.model_info.model_type_id} model type")
+
+    request = service_pb2.ListModelTypesRequest(user_app_id=self.user_app_id,)
+    response = self._grpc_request(self.STUB.ListModelTypes, request)
+    if response.status.code != status_code_pb2.SUCCESS:
+      raise Exception(response.status)
+    params = response_to_model_params(
+        response=response, model_type_id=self.model_info.model_type_id, template=template)
+    #yaml file
+    with open(yaml_file, 'w') as f:
+      yaml.dump(params, f, default_flow_style=False, sort_keys=False)
+    #updating the global model params
+    self.training_params.update(params)
+
+    return params
+
+  def update_params(self, **kwargs) -> None:
+    """Updates the model params for the model.
+
+    Args:
+        **kwargs: model params to update.
+
+    Example:
+        >>> from clarifai.client.model import Model
+        >>> model = Model(model_id='model_id', user_id='user_id', app_id='app_id')
+        >>> model_params = model.get_params(template='template', yaml_file='model_params.yaml')
+        >>> model.update_params({param1:value1, param2:value2})
```

Comment:
```suggestion
        >>> model.update_params({batch_size:8, dataset_version:v2})
```
I think having some example params helps